File: placement-reactnative/storyly-placement-react-native/android/src/main/java/com/storylyplacementreactnative/common/SPPlacementEventType.kt
package com.storylyplacementreactnative.common


enum class SPPlacementEventType(val eventName: String) {
    ON_WIDGET_READY("onWidgetReady"),
    ON_FAIL("onFail"),
    ON_EVENT("onEvent"),
    ON_ACTION_CLICKED("onActionClicked"),
    ON_PRODUCT_EVENT("onProductEvent"),
    ON_UPDATE_CART("onUpdateCart"),
    ON_UPDATE_WISHLIST("onUpdateWishlist"),
}

File: placement-flutter/storyly_placement_flutter/ios/Classes/Common/SPPlacementEventType.swift
import Foundation

@objc public enum SPPlacementEventType: Int {
  case onWidgetReady
  case onFail
  case onVisibilityChange
  case onEvent
  case onActionClicked
  case onProductEvent
  case onUpdateCart
  case onUpdateWishlist
  
  public var eventName: String {
    switch self {
    case .onWidgetReady:
      return "onWidgetReady"
    case .onFail:
      return "onFail"
    case .onVisibilityChange:
      return "onVisibilityChange"
    case .onEvent:
      return "onEvent"
    case .onActionClicked:
      return "onActionClicked"
    case .onProductEvent:
      return "onProductEvent"
    case .onUpdateCart:
      return "onUpdateCart"
    case .onUpdateWishlist:
      return "onUpdateWishlist"
    }
  }
  
  static var allCases: [SPPlacementEventType] {
    return [.onWidgetReady, .onFail, .onVisibilityChange, .onEvent, .onActionClicked, .onProductEvent, .onUpdateCart, .onUpdateWishlist]
  }
}

